cpp/ql/src/semmle/code/cpp/dataflow/internal/FlowVar.qll

```diff
@@ -147,10 +141,7 @@ private module PartialDefinitions {
      * - `inner` = `... .x`, `outer` = `&...`
      * - `inner` = `a`, `outer` = `*`
      */
-    predicate definesExpressions(Expr inner, Expr outer) {
-      inner = innerDefinedExpr and
-      outer = this
-    }
+    predicate definesExpressions(Expr inner, Expr outer) { none() }
```

Comment:
Can these four predicates be `abstract`? Then it should be more clear that they are always overridden.